lib/rust/frp/src/future.rs

```diff
@@ -0,0 +1,99 @@
+//! A module containing helpers for FRP networks in asynchronous environments.
+
+use crate::prelude::*;
+
+use crate::extend;
+use crate::node;
+use crate::stream::EventOutput;
+use crate::HasLabel;
+use crate::Label;
+use crate::Network;
+
+use std::pin::Pin;
+use std::task::Context;
+use std::task::Poll;
+
+
+
+// ===================
+// === FutureEvent ===
+// ===================
+
+/// A structure returned from [`EventOutputExt::next_event`] method.
+///
+/// It's a future which is resolved once the given frp node emits an event. It also allows checking
+/// if the value is provided - a useful tool when testing components with FRP API.
+#[derive(Debug)]
+pub struct FutureEvent<Out> {
+    _network: Network,
+    label:    Label,
+    value:    Rc<RefCell<Option<Out>>>,
+    wakers:   Rc<RefCell<SmallVec<[std::task::Waker; 1]>>>,
```

Comment:
That's interesting. I was always sure that you only need to store a single waker and replace it on each `poll`.

Reply:
I'm not sure. In fact, the `poll` shan't be called before being wakened by waker (and during waking we remove the Wakers from this list). So in most situations there should be only one waker. However, I've tracked situations when Futures are polled a second time despite not being waked, thus I've made a list.

Reply:
Well, technically executor can `poll` tasks indefinitely, without making use of `Waker`. I think you encountered such a case where a future was moved to another async task in the executor and it called `poll` to update stored Waker. My understanding comes from https://rust-lang.github.io/async-book/02_execution/03_wakeups.html
Also, I've never seen a future implementation that stores multiple Wakers, so I guess our case is not much different. 

Of course, it's not a mistake in any case, I am just really curious about it.